src/StatusQ/Components/private/statusMessage/StatusMessageEmojiReactions.qml

```diff
@@ -0,0 +1,233 @@
+import QtQuick 2.3
+import QtQuick.Controls 2.13
+import QtGraphicalEffects 1.13
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+
+Item {
+    id: root
+
+    height: 20
+    width: childrenRect.width
+
+    property int imageMargin: 4
+    signal addEmojiClicked(var sender, var mouse)
+    signal hoverChanged(bool hovered)
+    signal toggleReaction(int emojiID)
+
+    property var store
+    property bool isCurrentUser
+    property var emojiReactionsModel
+
+    property var icons: []
+
+    QtObject {
+        id: d
+
+        function lastTwoItems(nodes) {
+            return nodes.join(qsTr(" and "));
+        }
+
+        function showReactionAuthors(jsonArrayOfUsersReactedWithThisEmoji, emojiId) {
+            const listOfUsers = JSON.parse(jsonArrayOfUsersReactedWithThisEmoji)
+            if (listOfUsers.error) {
+                console.error("error parsing users who reacted to a message, error: ", obj.error)
+                return
+            }
+
+            let tooltip;
+            if (listOfUsers.length === 1) {
+                tooltip = listOfUsers[0]
+            } else if (listOfUsers.length === 2) {
+                tooltip = lastTwoItems(listOfUsers);
+            } else {
+                var leftNode = [];
+                var rightNode = [];
+                const maxReactions = 12
+                let maximum = Math.min(maxReactions, listOfUsers.length)
+
+                if (listOfUsers.length > maxReactions) {
+                    leftNode = listOfUsers.slice(0, maxReactions);
+                    rightNode = listOfUsers.slice(maxReactions, listOfUsers.length);
+                    return (rightNode.length === 1) ?
+                                lastTwoItems([leftNode.join(", "), rightNode[0]]) :
+                                lastTwoItems([leftNode.join(", "), qsTr("%1 more").arg(rightNode.length)]);
+                }
+
+                leftNode = listOfUsers.slice(0, maximum - 1);
+                rightNode = listOfUsers.slice(maximum - 1, listOfUsers.length);
+                tooltip = lastTwoItems([leftNode.join(", "), rightNode[0]])
+            }
+
+            tooltip += qsTr(" reacted with ");
```

Comment:
No :( Please don't glue the text like this, use `qsTr("%1 reacted with %2).arg(foo).arg(bar)`